array.go

```diff
@@ -2,43 +2,47 @@ package amf0
 
 import (
 	"bytes"
+	"encoding/binary"
 	"errors"
 	"io"
+	"reflect"
 )
 
 type Array struct {
-	*paired
+	*Paired
 }
 
-var _ AmfType = &Array{}
-var _ Paired = &Array{}
+var _ AmfType = new(Array)
 
 func NewArray() *Array {
-	return &Array{newPaired()}
+	return &Array{NewPaired()}
 }
 
+// Implements AmfType.Marker
+func (a *Array) Marker() byte { return 0x08 }
+
+func (a *Array) Native() reflect.Type { return reflect.TypeOf(a).Elem() }
+
 // Implements AmfType.Decode
 func (a *Array) Decode(r io.Reader) error {
-	numBytes, err := readBytes(r, 4)
-	if err != nil {
+	var n [4]byte
+	if _, err := r.Read(n[:]); err != nil {
```

Comment:
Bear in mind that read is not guaranteed to return enough bytes to fill up the buffer in one call! You can use [readAtLeast](https://golang.org/pkg/io/#ReadAtLeast) for this. 

My original readBytes function was written without realizing there was a function for this in the standard library :stuck_out_tongue: